MiniApp/Classes/core/RealMiniApp.swift

```diff
@@ -205,13 +232,78 @@ internal class RealMiniApp {
+                completionHandler(.failure(error))
+            }
+        }
     }
 
     /// Method to remove the unused/deleted items from the Keychain
     func cleanUpKeychain() {
         self.miniAppStatus.removeUnusedCustomPermissions()
     }
+
+    /// Method to check if all the required permissions mentioned in the manifest.json is agreed by the user.
+    /// - Parameters:
+    ///   - appId: MiniApp ID
+    ///   - versionId: Specific VersionID of a MiniApp
+    ///   - completionHandler: Handler that returns whether user agreed to required permissions or not.
+    func isRequiredPermissionsAllowed(appId: String, versionId: String, completionHandler: @escaping (Result<Bool, MASDKError>) -> Void) {
+        let cachedMetaData = self.miniAppManifestStorage.getManifestInfo(forMiniApp: appId)
+        if cachedMetaData?.version != versionId {
+            retrieveMiniAppMetaData(appId: appId, version: versionId) { (result) in
+                switch result {
+                case .success(let manifest):
+                    self.verifyRequiredPermissions(appId: appId,
+                                                   versionId: versionId,
+                                                   miniAppManifest: manifest,
+                                                   completionHandler: completionHandler)
+                case .failure(let error):
+                    completionHandler(.failure(error))
+                }
+            }
+        } else {
+            self.verifyRequiredPermissions(appId: appId,
+                                           versionId: versionId,
+                                           miniAppManifest: cachedMetaData?.miniAppManifest,
+                                           completionHandler: completionHandler)
+        }
+    }
+
+    /// Method that compares the required permissions from the manifest and the stored custom permissions.
+    /// - Parameters:
+    ///   - appId: MiniApp ID
+    ///   - requiredPermissions: List of required Custom permissions that is defined by the Mini App
+    ///   - completionHandler: Handler that returns whether user agreed to required permissions or not.
+    func verifyRequiredPermissions(appId: String,
+                                   versionId: String,
+                                   miniAppManifest: MiniAppManifest?,
+                                   completionHandler: @escaping (Result<Bool, MASDKError>) -> Void) {
+        guard let manifestData = miniAppManifest, let requiredPermissions = manifestData.requiredPermissions else {
+            return completionHandler(.success(true))
+        }
+        let storedCustomPermissions = self.miniAppPermissionStorage.getCustomPermissions(forMiniApp: appId)
+        let filtered = storedCustomPermissions.filter {
+            requiredPermissions.contains($0)
+        }
+        if filtered.count == requiredPermissions.count && filtered.allSatisfy({ $0.isPermissionGranted.boolValue == true }) {
```

Comment:
I think we also need to update ` MiniAppPermissionStorage` somewhere in this function - i.e. filter the stored custom permissions so it only has the entries from `manifest.json`. 

I think I had mentioned doing this after verifying the required permissions, but maybe it makes more sense to do this just before this `if` check?

Reply:
Ok, I made use of the filterCustomPermissions method which filters and the same is stored now